packages/api/src/controllers/clip.ts

```diff
@@ -61,49 +58,35 @@ app.use(
 );
 
 async function getProcessingClipsByRequesterId(
-  requesterId: string
-): Promise<WithID<Asset>[]> {
-  const createdAfter = Date.now() - 10 * 60 * 1000;
-  const assets = await db.asset.find([
-    sql`data->'status'->>'phase' IN ('waiting', 'processing', 'running')`,
-    sql`data->'source'->>'requesterId' = ${requesterId}`,
-    sql`coalesce((data->>'createdAt')::bigint, 0) > ${createdAfter}`,
+  requesterId: string,
+  ownerId: string
+): Promise<WithID<Task>[]> {
+  const scheduledAt = Date.now() - 5 * 60 * 1000;
+  const [tasks] = await db.task.find([
+    sql`data->>'userId' = ${ownerId}`,
+    sql`coalesce((data->>'scheduledAt')::bigint, 0) > ${scheduledAt}`,
   ]);
+  let runningClips: WithID<Task>[] = [];
+  const tasksByRequester = tasks.filter((t) => t.requesterId === requesterId);
+  tasksByRequester.forEach((task) => {
```

Comment:
Btw we should either write this iteratively (`forEach`, or even better a literate `for (.. of ..)`), or functionally `filter`. Ideally do not mix the 2 patterns otherwise it's just a back-and-forth to read through it.

Using the functional approach (generally my preference unless it's too complex of a pipeline), you would just chain 2 or more `.filter()` calls with conditions for which to filter the tasks.